### ISSUE TITLE
Regressions in windows arm/arm64 hardware intrinsics tests.

### ISSUE THREAD

**Issue by sandreenko** [dotnet/coreclr#20149]
There are 6 tests that regressed last week.
The last green job was [Sep 19](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_windows_nt_innerloop_tst_prtest/1619/). 

Then it was failing because of infrastructure problems and now it fails with 6 failed tests.

Failing tests:
```
JIT\HardwareIntrinsics\X86\Sse2\SumAbsoluteDifferences_ro\SumAbsoluteDifferences_ro.cmd
JIT\HardwareIntrinsics\X86\Popcnt\Popcnt_r\Popcnt_r.cmd
JIT\HardwareIntrinsics\X86\Sse2\MultiplyHorizontalAdd_r\MultiplyHorizontalAdd_r.cmd
JIT\HardwareIntrinsics\X86\Sse2\MultiplyHorizontalAdd_ro\MultiplyHorizontalAdd_ro.cmd
JIT\HardwareIntrinsics\X86\Popcnt\Popcnt_ro\Popcnt_ro.cmd
JIT\HardwareIntrinsics\X86\Sse2\SumAbsoluteDifferences_r\SumAbsoluteDifferences_r.cmd
```

They all fail with:
```
TEST_CMD_LINE: JIT\HardwareIntrinsics\X86\Sse2\SumAbsoluteDifferences_ro> JIT\HardwareIntrinsics\X86\Sse2\SumAbsoluteDifferences_ro\SumAbsoluteDifferences_ro.cmd 

No output was generated!

TEXT_EXPECTED_RETURN_CODE = 0
TEXT_ACTUAL_RETURN_CODE = -1 (0xffffffff)

```

[Example](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_windows_nt_innerloop_tst_prtest/1638/).

Note: there are also jobs that fail with "Unable to find the appropriate revision to build. Merge conflicts? If so, resolve them and update PR. Has the branch been deleted from the repo? If yes, the create a PR to remove the branch from https://github.com/dotnet/dotnet-ci/blob/master/data/repolist.txt. If this is a ROSLYN branch: Feature branch? If yes, then this failure is expected - we do not currently support running internal jobs against feature branches. Future, master or stabilization? If yes, then its likely the commit pump between Roslyn to Roslyn internal is not running. Contact dnceng@microsoft.com. " and it is a different problem, [example](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_windows_nt_innerloop_tst_prtest/1659/).

**Comment by sandreenko** [comment]
cc @dotnet/jit-contrib.

**Comment by 4creators** [comment]
@sandreenko Looking into it.

**Comment by 4creators** [comment]
Regression was caused by  PR `Fix inconsistent Intel hardware intrinsic APIs` #19949 cc @fiigii @tannergooding 

The reason for it is that the above tests were removed while they still remained on exclusion lists what leads to errors. There is an issue which tracks this behavior of Windows arm/arm64 infrastructure: #19882.

**Comment by 4creators** [comment]
@fiigii @tannergooding Do you intend to keep the above tests disabled or some of them could be reenabled? I am working on fix for this issue.

**Comment by fiigii** [comment]
I disabled them in that PR. Do we need more work for arm/arm64?

**Comment by fiigii** [comment]
> The reason for it is that the above tests were removed while they still remained on exclusion lists what leads to errors. 

Oh, I see, thanks.

**Comment by 4creators** [comment]
@fiigii Do you think that we can reenable `SumAbsoluteDifferences` and `MultiplyHorizontalAdd` or do you plan to change these APIs?

**Comment by fiigii** [comment]
I will write new test cases (template-based) for them. But CI is still stuck, and we need the API change flow back from CoreFX.

**Comment by 4creators** [comment]
OK than I will drop them from Test.lst what will fix CI. You would need to add back new tests to the list to keep CI clean when they are implemented.

**Comment by fiigii** [comment]
@4creators Thank you so much for the help!

**Comment by 4creators** [comment]
@sandreenko Fix for the issue is in #20151

**Comment by sandreenko** [comment]
Thank you @4creators!
